### ISSUE TITLE
RaytracingHelloWorld crash in m_fallbackDevice->CreateStateObject()

### ISSUE THREAD

**Issue by giordi91** [Microsoft/DirectX-Graphics-Samples#458]
I am currently testing the DXR and fallbacklayer, both in my standalone application and the given samples. I am able to run the samples properly on a surface pro 2017 on Intel graphics and on an Intel + GTX 1080 in my home build. 

Unluckily I am getting a crash in the samples as-well on my work machine  which runs an 8700k + dual titan Xp not in SLI. I am trying to figure out what the difference can be to try to solve the crash, I thought I would open an issue here in hope to get any hint. 
Please find below a stack trace and the output from VS.

![output](https://user-images.githubusercontent.com/2525797/45497623-33518200-b770-11e8-9017-61f788a748b6.jpg)
![callstack](https://user-images.githubusercontent.com/2525797/45497625-3482af00-b770-11e8-838f-a462dc67997e.jpg)

The crash is happening at:
https://github.com/Microsoft/DirectX-Graphics-Samples/blob/f8b7a6141c4adc3128b359c3d373129cb5728f55/Samples/Desktop/D3D12Raytracing/src/D3D12RaytracingHelloWorld/D3D12RaytracingHelloWorld.cpp#L281

I have tried to update the driver to the latest aswell and still get exactly the same crash.
Currently running:
NVidia driver: 399.24
Visual Studio : 15.8.2, SDK: 10.0.17134.0

Any question or any way I can help please let me know.

**Comment by giordi91** [comment]
I am 99.9% sure, I have downloaded the source from master this morning, I tried with both 1.5 and 1.3 binaries (1.5 doesn't work, they are recognized as stale). Developer mode is on, graphics tools are installed and working. Windows build is 1803 and up to date. The only thing that comes to mind was that we are behind a firewall, and we struggled a little with the updates, that in the end we opened the gates and got all the updates and installed the graphics tools, not sure if something might have gone wrong there. Regular dx12 application work even in debug. I will try again tomorrow morning, everything clean and get back to you.

to note, the other CreateStateObject seems to work fine. I will try again, I am also packaging the working solution I have at home and try that one at work as well.

**Comment by giordi91** [comment]
Good afternoon, I re-did all the steps, made sure of all the requirements, and unluckily still the same issue, the only thing that looks a bit weird is developer mode, where I get this : 
![devmode](https://user-images.githubusercontent.com/2525797/45555576-0fa24080-b831-11e8-8d9c-c0f87c28f34a.png)

Mostly because behind firewall. I don't think should be an issue but we are getting it fixed today anyway. 
In parallel we are working on another option, updating a colleague of mine machine which runs on 1080tis instead of Titans Xps. If works on his machine might be a driver issue, if does not work might be our setup. If the latter, we will try to have a machine wiped clean, disconnected from internal network and do a clean install with internet enabled and see if we get the same issue. 

In short for now the problem persists. If you get any idea or test we can do please let me know.

**Comment by giordi91** [comment]
I have an update, developer mode now is installed nice and clean, we still got the crash.
We updated my co-worker machine which runs dual 1080ti and worked no issue. I then proceeded in butchering another machine and taking out a 1080ti and put it on my system instead of the titans, just to be safe, I did a clean driver install. The issues is still there. 

At this point it clearly seems an issue of my system, not hardware related. All the updates from windows are cleanly installed, and all other requirements are met. Is there anything that I can do to try to debug this? Any diagnostic windows too that might help? I would rather not nuke my machine and rebuild a windows environment from scratch, but we will if we have to.

Cheers

**Comment by pkristof** [comment]
Hi, can you copy&paste the whole debug spew log for the sample?

**Comment by pkristof** [comment]
The debug spew shows the right GPU adapter being picked and Fallback Layer enabled, so that's good. Your setup is expected to work as you verified on other machines and I did as well for the GPU and the driver. Since you already redid all the setup steps, I'm afraid I don't have any other further suggestions at this moment. You could dual boot another OS instance/VHD and see if that clean environment gets you further before completely flattening the primary OS instance.

**Comment by giordi91** [comment]
Apologies for the delay, we managed to make it to work after a clean re-install. We are seeing a different crash in a different place for the mini-engine, I will open a different issue. Cheers.
